src/test-run/commands/from-object.js

```diff
@@ -36,88 +37,91 @@ import { WaitCommand, DebugCommand } from './observation';
+            return new SelectTextCommand(obj, testRun);
 
         case TYPE.selectTextAreaContent:
-            return new SelectTextAreaContentCommand(obj);
+            return new SelectTextAreaContentCommand(obj, testRun);
 
         case TYPE.selectEditableContent:
-            return new SelectEditableContentCommand(obj);
+            return new SelectEditableContentCommand(obj, testRun);
 
         case TYPE.pressKey:
-            return new PressKeyCommand(obj);
+            return new PressKeyCommand(obj, testRun);
 
         case TYPE.wait:
-            return new WaitCommand(obj);
+            return new WaitCommand(obj, testRun);
 
         case TYPE.navigateTo:
-            return new NavigateToCommand(obj);
+            return new NavigateToCommand(obj, testRun);
 
         case TYPE.setFilesToUpload:
-            return new SetFilesToUploadCommand(obj);
+            return new SetFilesToUploadCommand(obj, testRun);
 
         case TYPE.clearUpload:
-            return new ClearUploadCommand(obj);
+            return new ClearUploadCommand(obj, testRun);
 
         case TYPE.takeScreenshot:
-            return new TakeScreenshotCommand(obj);
+            return new TakeScreenshotCommand(obj, testRun);
 
         case TYPE.takeElementScreenshot:
-            return new TakeElementScreenshotCommand(obj);
+            return new TakeElementScreenshotCommand(obj, testRun);
 
         case TYPE.resizeWindow:
-            return new ResizeWindowCommand(obj);
+            return new ResizeWindowCommand(obj, testRun);
 
         case TYPE.resizeWindowToFitDevice:
-            return new ResizeWindowToFitDeviceCommand(obj);
+            return new ResizeWindowToFitDeviceCommand(obj, testRun);
 
         case TYPE.maximizeWindow:
-            return new MaximizeWindowCommand(obj);
+            return new MaximizeWindowCommand(obj, testRun);
 
         case TYPE.switchToIframe:
-            return new SwitchToIframeCommand(obj);
+            return new SwitchToIframeCommand(obj, testRun);
 
         case TYPE.switchToMainWindow:
             return new SwitchToMainWindowCommand();
 
         case TYPE.setNativeDialogHandler:
-            return new SetNativeDialogHandlerCommand(obj);
+            return new SetNativeDialogHandlerCommand(obj, testRun);
 
         case TYPE.setTestSpeed:
-            return new SetTestSpeedCommand(obj);
+            return new SetTestSpeedCommand(obj, testRun);
 
         case TYPE.setPageLoadTimeout:
-            return new SetPageLoadTimeoutCommand(obj);
+            return new SetPageLoadTimeoutCommand(obj, testRun);
 
         case TYPE.assertion:
-            return new AssertionCommand(obj);
+            return new AssertionCommand(obj, testRun);
 
         case TYPE.debug:
-            return new DebugCommand(obj);
+            return new DebugCommand(obj, testRun);
+
+        case TYPE.executeExpression:
+            return new ExecuteExpressionCommand(obj, testRun);
```

Comment:
May be better to extract `getCommandCtor` method and after call it with`obj, testRun` args